SIMD.py

```diff
@@ -310,87 +310,4 @@ def IntegerPowSIMD(a, n):
-    for item in preorder_traversal(expr):
-        if item.func == RationalTMP:
-            # Set variable name
-            if item.args[0] * item.args[1] < 0:
-                SIMD_const_varnms.extend(["_Rational_m" + str(abs(item.args[0])) + "_" + str(abs(item.args[1])) + SIMD_const_suffix])
-            elif item.args[0] > 0 and item.args[1] > 0:
-                SIMD_const_varnms.extend(["_Rational_" + str(item.args[0]) + "_" + str(item.args[1]) + SIMD_const_suffix])
-            else:
-                # E.g., doesn't make sense to have -1/-3. SymPy should have simplified this.
-                print("Found a weird Rational(a, b) expression, where a < 0 and b < 0. Report to SymPy devels.")
-                print("Specifically, found that a = " + str(item.args[0]) + " and b = " + str(item.args[1]))
-                sys.exit(1)
-            # Set variable value, to 34 digits of precision
-            SIMD_const_values.extend([str(N(Float(item.args[0], 34)/Float(item.args[1], 34), 34))])
-        elif item.func == IntegerTMP:
-            # Set variable name
-            if item.args[0] < 0:
-                SIMD_const_varnms.extend(["_Integer_m" + str(-item.args[0]) + SIMD_const_suffix])
-            else:
-                SIMD_const_varnms.extend(["_Integer_" + str(item.args[0]) + SIMD_const_suffix])
-            # Set variable value, to 34 digits of precision
-            SIMD_const_values.extend([str((Float(item.args[0], 34)))])
-
-    # Step 6b: Replace all integers and rationals with the appropriate variable names:
-    for item in preorder_traversal(expr):
-        tempitem = item
-        if item.func == RationalTMP:
-            if item.args[0] * item.args[1] < 0:
-                tempitem = var("_Rational_m" + str(abs(item.args[0])) + "_" + str(abs(item.args[1])) + SIMD_const_suffix)
-            elif item.args[0] > 0 and item.args[1] > 0:
-                tempitem = var("_Rational_" + str(item.args[0]) + "_" + str(item.args[1]) + SIMD_const_suffix)
-            else:
-                # E.g., doesn't make sense to have -1/-3. SymPy should have simplified this.
-                print("Found a weird Rational(a, b) expression, where a < 0 and b < 0. Report to SymPy devels.")
-                print("Specifically, found that a = " + str(item.args[0]) + " and b = " + str(item.args[1]))
-                sys.exit(1)
-        elif item.func == IntegerTMP:
-            if item.args[0] < 0:
-                tempitem = var("_Integer_m" + str(-item.args[0]) + SIMD_const_suffix)
-            else:
-                tempitem = var("_Integer_" + str(item.args[0]) + SIMD_const_suffix)
-        if item != tempitem: expr = expr.subs(item, tempitem)
-
-    def lookup_name_output_idx(name, list_of_names):
-        for i in range(len(list_of_names)):
-            if list_of_names[i] == name:
-                return i
-        print("I SHOULDN'T BE HERE!", name, list_of_names)
-        sys.exit(1)
-
-    if debug == "True":
```

Comment:
Let's restore the debug mode. It should work even better now that we've sanitized the integers & rationals out of the input expression. Basically the debug mode un-does the expr->SIMD (i.e., SIMD->expr), then computes the simplified difference between the output expr and the input expr. The difference should be zero.